packages/client/src/extension.mts

```diff
@@ -191,11 +200,42 @@ async function _activate(options: ActivateOptions): Promise<ExtensionApi> {
         createLanguageStatus({ areIssuesVisible: () => decorator.visible, onDidChangeVisibility: decorator.onDidChangeVisibility }),
         registerActionsMenu({ areIssuesVisible: () => decorator.visible }),
         client.onBlockFile(notifyUserOfBlockedFile),
+        /** Listen to document changes to trigger autocorrect. */
+        vscode.workspace.onDidChangeTextDocument(handleOnDidChangeTextDocument),
     );
 
     performance.mark('registerCspellInlineCompletionProviders');
     await registerCspellInlineCompletionProviders(context.subscriptions).catch(() => undefined);
 
+    function handleOnDidChangeTextDocument({ document, contentChanges }: vscode.TextDocumentChangeEvent) {
+        // TODO: Currently we don't wait for suggestions that are still being generated when this handler runs.
+        // We may want to consider waiting for these in the future.
+
+        // Check if autocorrect is enabled.
+        if (!vscode.workspace.getConfiguration().get('cSpell.autocorrect', false)) return;
+
+        // Check if the change should trigger autocorrect.
+        if (contentChanges.length === 0) return;
+        const lastChange = contentChanges[contentChanges.length - 1];
+        if (!AUTO_CORRECT_TRIGGER.test(lastChange.text)) return;
+
+        // Get cursor position prior to the change.
+        const position = lastChange.range.start;
+        const prevPosition = position.character > 0 ? position.translate(0, -1) : undefined;
+        if (!prevPosition) return;
+
+        // Check if the cursor was at the end of a word.
+        const wordRange = document.getWordRangeAtPosition(prevPosition);
```

Comment:
Some edge conditions to think of:
- What if the person typed: `FirstCafëTerrace. `? 
- What if the person typed: `FirstCafëTerrace.⏎`? `⏎` is return.

Would they get picked up?